src/hooks/useSearch.ts

```diff
@@ -1,14 +1,50 @@
-import { UserRepository } from '@amityco/ts-sdk-react-native';
+import {
+  CommunityRepository,
+  UserRepository,
+} from '@amityco/ts-sdk-react-native';
 import { useCallback, useEffect, useState, useRef } from 'react';
 import { ISearchItem } from '../components/SearchItem';
 
 export type TSearchItem = Amity.User &
-  ISearchItem & { name: string; id: string };
+  ISearchItem &
+  Amity.Membership<'community'> & { name: string; id: string };
 
-const useSearch = (searchText: string = '') => {
+const useSearch = (
+  searchText: string = '',
+  privateCommunityId: string = ''
+) => {
   const onNextPageRef = useRef<(() => void) | undefined | null>(null);
   const [searchResult, setSearchResult] = useState<TSearchItem[]>([]);
-  const searchData = useCallback((text: string) => {
+  const searchPrivateCommunityMember = useCallback(
+    (text: string) => {
+      return CommunityRepository.Membership.getMembers(
+        {
+          communityId: privateCommunityId,
+          search: text,
+          limit: 5,
+          sortBy: 'firstCreated',
+          memberships: ['member'],
+        },
+        ({ data, error, hasNextPage, onNextPage }) => {
+          if (error) return null;
+          hasNextPage
+            ? (onNextPageRef.current = onNextPage)
+            : (onNextPageRef.current = null);
```

Comment:
Can we use this instead ?

```suggestion
          onNextPageRef.current = hasNextPage ? onNextPage : null;
```

Reply:
Done krub